### PR TITLE
macOS fix selected Group text color

### PR DESCRIPTION
**Pull request by sweetppro**
Fixes: #4806

Notes: Fixed `4.0.0` bug that didn't highlight the current selection in `View > Use Groups`.

### CODE REVIEW

File: macosx/GroupTextCell.mm [hunk 1]
```diff
@@ -22,9 +23,15 @@ - (void)drawInteriorWithFrame:(NSRect)cellFrame inView:(NSView*)controlView
...
     NSRect titleRect = [self titleRectForBounds:cellFrame];
     NSMutableAttributedString* string = [[self attributedStringValue] mutableCopy];
-    NSDictionary* attributes = [NSDictionary
+    NSMutableDictionary* attributes = [NSMutableDictionary
         dictionaryWithObjects:@[ [NSFont boldSystemFontOfSize:11.0], [NSColor secondaryLabelColor] ]
                       forKeys:@[ NSFontAttributeName, NSForegroundColorAttributeName ]];
+
+    if (self.selected)
+    {
+        [attributes setObject:[NSColor labelColor] forKey:NSForegroundColorAttributeName];
```
Comment (Coeur) [inline macosx/GroupTextCell.mm:32]: nit: modern obj-c could be used.

Comment (Coeur) [reply macosx/GroupTextCell.mm:32]: You're actually setting NSForegroundColorAttributeName twice, so I'd prefer to  go for a ternary pattern or an `if {} else {}` pattern.

Example:
```
NSDictionary* attributes = [NSDictionary
         dictionaryWithObjects:@[ [NSFont boldSystemFontOfSize:11.0], selected ? [NSColor labelColor] : [NSColor secondaryLabelColor] ]
                       forKeys:@[ NSFontAttributeName, NSForegroundColorAttributeName ]];
```

File: macosx/TorrentTableView.mm [hunk 2]
```diff
@@ -221,6 +221,22 @@ - (void)outlineView:(NSOutlineView*)outlineView
...
+            GroupTextCell* groupCell = cell;
+            groupCell.selected = NO;
+
+            // if cell is selected, set selected flag so we can style the title in GroupTextCell drawInteriorWithFrame
+            NSInteger const row = [self rowForItem:item];
+            NSIndexSet* selectedRowIndexes = self.selectedRowIndexes;
+            if ([selectedRowIndexes containsIndex:row])
+            {
+                groupCell.selected = YES;
+            }
```
Comment (Coeur) [inline macosx/TorrentTableView.mm:237]: You're setting `selected` twice (already on line 229).
So best to directly do unconditionally:
```
groupCell.selected = [selectedRowIndexes containsIndex:row];
```

**Comment by Coeur** [comment]
UI-wise, the change is good. So only some nit code-style to address and it's good.

File: macosx/TorrentTableView.mm (hunk 2 above)
Comment (sweetppro) [reply macosx/TorrentTableView.mm]: done, thanks

File: macosx/GroupTextCell.mm (hunk 1 above)
Comment (sweetppro) [reply macosx/GroupTextCell.mm]: done

**Pull request merged**